src/frontend/JasmineGraphFrontEnd.cpp

```diff
@@ -943,18 +949,18 @@ static void add_graph_cust_command(std::string masterIP, int connFd, SQLiteDBInt
             name + "\", \"" + edgeListPath + "\", \"" + uploadStartTime + "\", \"\",\"" +
             to_string(Conts::GRAPH_STATUS::LOADING) + "\", \"\", \"\", \"\")";
         int newGraphID = sqlite->runInsert(sqlStatement);
-        MetisPartitioner *partitioner = new MetisPartitioner(sqlite);
+        MetisPartitioner partitioner(sqlite);
         vector<std::map<int, string>> fullFileList;
-        partitioner->loadContentData(attributeListPath, graphAttributeType, newGraphID, attrDataType);
-        partitioner->loadDataSet(edgeListPath, newGraphID);
-        int result = partitioner->constructMetisFormat(Conts::GRAPH_TYPE_NORMAL);
+        partitioner.loadContentData(attributeListPath, graphAttributeType, newGraphID, attrDataType);
+        partitioner.loadDataSet(edgeListPath, newGraphID);
+        int result = partitioner.constructMetisFormat(Conts::GRAPH_TYPE_NORMAL);
         if (result == 0) {
-            string reformattedFilePath = partitioner->reformatDataSet(edgeListPath, newGraphID);
-            partitioner->loadDataSet(reformattedFilePath, newGraphID);
-            partitioner->constructMetisFormat(Conts::GRAPH_TYPE_NORMAL_REFORMATTED);
-            fullFileList = partitioner->partitioneWithGPMetis("");
+            string reformattedFilePath = partitioner.reformatDataSet(edgeListPath, newGraphID);
+            partitioner.loadDataSet(reformattedFilePath, newGraphID);
+            partitioner.constructMetisFormat(Conts::GRAPH_TYPE_NORMAL_REFORMATTED);
+            fullFileList = partitioner.partitioneWithGPMetis("");
         } else {
-            fullFileList = partitioner->partitioneWithGPMetis("");
+            fullFileList = partitioner.partitioneWithGPMetis("");
```

Comment:
There is no need of `else` here. The following code,

```
        if (result == 0) {
            string reformattedFilePath = partitioner.reformatDataSet(edgeListPath, newGraphID);
            partitioner.loadDataSet(reformattedFilePath, newGraphID);
            partitioner.constructMetisFormat(Conts::GRAPH_TYPE_NORMAL_REFORMATTED);
            fullFileList = partitioner.partitioneWithGPMetis("");
        } else {
            fullFileList = partitioner.partitioneWithGPMetis("");
        }
```

Can be refactored as follows,

```
        if (result == 0) {
            string reformattedFilePath = partitioner.reformatDataSet(edgeListPath, newGraphID);
            partitioner.loadDataSet(reformattedFilePath, newGraphID);
            partitioner.constructMetisFormat(Conts::GRAPH_TYPE_NORMAL_REFORMATTED);
        }
            fullFileList = partitioner.partitioneWithGPMetis("");
```

Reply:
Done

Reply:
Ok